### DISCUSSION TITLE
Ma l'assistente vocale integrato... Capisce quello che dite?

### DISCUSSION THREAD

**Discussion by Pippisbruffi** [indomus/forum#3140] [HUB: Home Assistant]
Ciao a tutti, chiedo un aiuto. Recentemente ho creato un esp32 con un microfono INMP441, un piccolo speaker e un piccolo display per utilizzarlo con la nuova funzione "voice assist". Nella sezione assistenti vocali, nelle impostazioni di home assistant, utilizzo "home assistant cloud" sia come riconoscimento vocale che come sintesi vocale.

Funziona tutto, compresa la wake-word, però sto constatando che a livello di riconoscimento vocale non ne azzecca una. Grazie ai log di esphome, posso vedere cosa ha capito il dispositivo tramite il microfono.
Per fare due esempi, ho appena detto "accendi comodino" e lui ha capito "a gente comodino", oppure se dico frasi troppo lunghe me le tronca a metà, per esempio se dico "stato temperatura cucina" lui capisce solo "stato temperatura".
Inoltre sto riscontrando problemi con le entità, per esempio ho un termometro chiamato "temperatura camera". Quando gli chiedo "temperatura camera" o frasi simili, anche se capisce correttamente quello che ho detto, mi risponde sempre "mi dispiace, non ho capito".
Mi chiedo se è un problema del mio setup o se effettivamente l'assistente vocale è al momento completamente inutilizzabile :')

Utilizzando openai (chatgpt) come provider le cose migliorano, nel senso che anche se nei log vedo comunque che ha capito "a gente comodino", lui effettivamente accende il comodino almeno. Comunque è lontanissimo dall'essere utilizzabile quotidianamente...

**Comment by angio89** [comment]
Ciao,
guardando diversi video su internet credo che il problema del riconoscimento vocale sia noto a tutti, soprattutto per quanto riguarda la lingua italiana.  Per fare dei test, prova ad utilizzare la lingua inglese e, al posto di "accendi/spegni", utilizza "attiva/disattiva". Ovviamente, con l'aiuto dell'IA, come ChatGPT, le cose migliorano molto per ovvi motivi. Mi sa che toccherà attendere ancora un po'...

**Comment by duedipicche** [comment]
Scrivo qui per non aprire un altro topic.
Ho messo homeassistant core su un raspberry pi 5 16 MB e tutto funziona a meraviglia.
Dopo diversi mesi volevo aggiungere alcuni comandi vocali, ma non volevo usa il cloud.
Ho installato wyoming-whisper wyoming-piper e wyoming-wakeword su home assistant core.
Le integrazioni sono state aggiunte e ho settato la configurazione di Assist.
Dimenticavo il tutto gira su docker.
Ora sono fermo ho attivato un microfono sul raspberry, funziona alla grande con vosk e con porcupine, ma non riesco a capire come integrare l'accensione di un relay attraverso un comando vocale da home assistant usando Assist.
Naturalmente ho messo gli alias e esposto lo swith a Assist.
Se qualcuno mi spiegasse o almeno mi mettesse un link ad un esempio ne sarei grato.
Emanuele
p.s. ho provato a "parlare" col raspberry, ma nulla accade

**Comment by smarthomeroero-web** [comment]
> Ciao a tutti, chiedo un aiuto. Recentemente ho creato un esp32 con un microfono INMP441, un piccolo speaker e un piccolo display per utilizzarlo con la nuova funzione "voice assist". Nella sezione assistenti vocali, nelle impostazioni di home assistant, utilizzo "home assistant cloud" sia come riconoscimento vocale che come sintesi vocale.
> 
> Funziona tutto, compresa la wake-word, però sto constatando che a livello di riconoscimento vocale non ne azzecca una. Grazie ai log di esphome, posso vedere cosa ha capito il dispositivo tramite il microfono. Per fare due esempi, ho appena detto "accendi comodino" e lui ha capito "a gente comodino", oppure se dico frasi troppo lunghe me le tronca a metà, per esempio se dico "stato temperatura cucina" lui capisce solo "stato temperatura". Inoltre sto riscontrando problemi con le entità, per esempio ho un termometro chiamato "temperatura camera". Quando gli chiedo "temperatura camera" o frasi simili, anche se capisce correttamente quello che ho detto, mi risponde sempre "mi dispiace, non ho capito". Mi chiedo se è un problema del mio setup o se effettivamente l'assistente vocale è al momento completamente inutilizzabile :')
> 
> Utilizzando openai (chatgpt) come provider le cose migliorano, nel senso che anche se nei log vedo comunque che ha capito "a gente comodino", lui effettivamente accende il comodino almeno. Comunque è lontanissimo dall'essere utilizzabile quotidianamente...

Ne approfitto per chiederti se hai uno schema hardware o se hai seguito un tutorial: io riscontro un audio distorto e con volume molto basso

**Reply by Pippisbruffi** [reply]
Ho seguito il tutorial del canale youtube everything smart home, ma é vecchio e alcune cose sono cambiate. Per il volume, adesso devi usare la stringa "gain_factor:" nella configurazione del microfono.
Per quanto riguarda la distorsione, non mi é mai capitato che distorcesse particolarmente, ma potrebbe esserci del rumore di fondo abbastanza forte dato da collegamenti poco isolati o da un microfono di scarsa qualitá... Prova ad accorciare i cavetti e a saldarli invece di collegarli con i dupont, magari migliora qualcosa

**Reply by smarthomeroero-web** [reply]
Grazie della rapida risposta. Il video dovrebbe essere questo: https://www.youtube.com/watch?v=zhlIaBG3Ldo
Faccio le modifiche che mi hai suggerito e incrocio le dita...
